onnxruntime/core/providers/cpu/ml/svmclassifier.cc

```diff
@@ -125,120 +153,71 @@ Status SVMClassifier<T>::Compute(OpKernelContext* ctx) const {
-          maxclass = k;
-          maxweight = scores[k];
-        }
-      }
+      auto it_maxweight = std::max_element(scores.begin(), scores.end());
+      maxclass = std::distance(scores.begin(), it_maxweight);
+      maxweight = *it_maxweight;
     }
-    //write top class
+
+    // write top class
+    // onnx specs expects one column per class.
     int write_additional_scores = -1;
-    if (rho_.size() == 1)  //binary
-    {
+    if (rho_.size() == 1) {
       if (using_strings_) {
-        if (classlabels_strings_.size() == 2 && weights_are_all_positive_ && maxweight >= 0.5 && proba_.size() == 0) {
-          Y->template MutableData<std::string>()[n] = classlabels_strings_[1];  //positive label
-          write_additional_scores = 0;
-        } else if (classlabels_strings_.size() == 2 && maxweight > 0 && !weights_are_all_positive_ && proba_.size() == 0) {
-          Y->template MutableData<std::string>()[n] = classlabels_strings_[1];  //positive label
-          write_additional_scores = 0;
-        } else if (classlabels_strings_.size() == 2 && proba_.size() > 0) {            //this case all classes are in their rightful spot
-          Y->template MutableData<std::string>()[n] = classlabels_strings_[maxclass];  //whichever label
-          write_additional_scores = -1;
-        } else if (classlabels_strings_.size() == 2) {
-          Y->template MutableData<std::string>()[n] = classlabels_strings_[0];  //negative label
-          write_additional_scores = 1;
-        } else if (maxweight > 0) {
-          Y->template MutableData<std::string>()[n] = "1";  //positive label
-        } else {
-          Y->template MutableData<std::string>()[n] = "0";  //negative label
-        }
-      } else  //no strings
-      {
-        if (classlabels_ints_.size() == 2 && weights_are_all_positive_ && maxweight >= 0.5 && proba_.size() == 0) {
-          Y->template MutableData<int64_t>()[n] = classlabels_ints_[1];  //positive label
-          write_additional_scores = 0;
-        } else if (classlabels_ints_.size() == 2 && maxweight > 0 && !weights_are_all_positive_ && proba_.size() == 0) {
-          Y->template MutableData<int64_t>()[n] = classlabels_ints_[0];  //pos  label
-          write_additional_scores = 0;
-        } else if (classlabels_ints_.size() == 2 && proba_.size() > 0)  //this case all classes are in their rightful spot
-        {
-          Y->template MutableData<int64_t>()[n] = classlabels_ints_[maxclass];  //whichever label
-          write_additional_scores = -1;
-        } else if (classlabels_ints_.size() == 2) {
-          Y->template MutableData<int64_t>()[n] = classlabels_ints_[0];  //negative label
-          write_additional_scores = 1;
-        } else if (maxweight > 0) {
-          Y->template MutableData<int64_t>()[n] = 1;  //positive label
-        } else {
-          Y->template MutableData<int64_t>()[n] = 0;  //negative label
-        }
+        SETSCORESVM(std::string, classlabels_strings_, "1", "0")
```

Comment:
Do we really want to write code in a weekly-typed way by using macro?

Reply:
Replaced by a template.